test/bundler/esbuild/default.test.ts

```diff
@@ -2797,14 +2799,16 @@ describe("bundler", () => {
   });
   itBundled("default/ImportMetaCommonJS", {
     files: {
-      "/entry.js": `console.log(import.meta.url, import.meta.path)`,
+      "/entry.js": `
+      import fs from "fs";
+      import { fileURLToPath } from "url";
+      console.log(fs.existsSync(fileURLToPath(import.meta.url)), fs.existsSync(import.meta.path));
+      `,
     },
     format: "cjs",
-    bundleWarnings: {
-      "/entry.js": [`"import.meta" is not available with the "cjs" output format and will be empty`],
-    },
+    target: "node",
     run: {
-      stdout: "undefined undefined",
+      stdout: "true true",
```

Comment:
@paperdave

Reply:
ooh i like this. `import.meta` *should* just work, aliasing to `__dirname` and `__filename`.

this test however should

- run in node to ensure `import.meta` is not actually emitted. you can do this by setting `runtime: "node"` in the run object.
- fs.existsSync is not a good enough behavior test. what if the path is `/`. it should be a string that is checked for exactly. if you turn the config object passed to itBundled into a function with signature `(({ root }) => ({ files: ...` you can use that path to write exact expectations for the absolute paths. see `tsconfig/Path` for more details on this api.

though after irl mention of this, it seems we *are* emitting import.meta, so maybe we just make this behavior error since that is likely easier.